### ISSUE TITLE
A more efficient CollectionView's Filter()

### ISSUE THREAD

**Issue by frankcortes** [marionettejs/backbone.marionette#2590]
Hi everyone, I'm a Marionette fan who enjoys your nice framework. My question is about the new 'filter()' functionality (http://marionettejs.com/docs/v2.4.1/marionette.collectionview.html#collectionviews-filter).

In your docs, you encourage me to use `render()` function to 'render again' the `CollectionView` when I have modified the filter to display the changes. However, this behaviour will render the whole `CollectionView` again, when I only need to remove or add some `childrenView`. Maybe an alternative function like this could be better that a simple `render`(in pseudocode):

```javascript
/*
if the filter is the same than before, return
else
   for each model in collectionView
      if filter(model) is true
         if  not existRepresentation(model,view)
           this._onCollectionAdd(model)
      else
         if existRepresentation(model, view)
           this._onCollectionRemove(model)
*/
```

What are your thoughts? Thanks in advance!

**Comment by cmaher** [comment]
I'd be hesitant to change how the render function works, since a lot of implicit behavior relies on that. However, in the [original PR](https://github.com/marionettejs/backbone.marionette/pull/2261), there was discussion around a possible `setFilter` method, which I think would be able to handle the more efficient rendering you propose.

**Comment by frankcortes** [comment]
I agree with you. In my view, this 'efficient rendering' shouldn't be directly in `render()` function. The expected functionality for `render()` is to remove the previous DOM and build the whole new structure according with its state. Maybe, an extra method like `setFilter` could be a great idea.

**Comment by cmaher** [comment]
A PR would be great, as long as someone else from @marionettejs/marionette-core thinks this should go in 2.x

**Comment by ianmstew** [comment]
I think this is a great idea.  `2.4.2` is wrapped up and at this point we are working toward v3.  @frankcortes open up a PR against `next`, the eventual v3, and if there ends up being a `2.4.3` then we can cherry-pick it.

**Comment by paulfalgout** [comment]
I'm currently hoping for a `2.5` but similarly I'd prefer this pulled againt `next` and if we need/want to cherry pick it for a `2.x` great.  Seems like a good addition.  The `minor` branch is kind of dead in the water at the moment.

**Comment by ianmstew** [comment]
@frankcortes Looking forward to the additions re: #2596.

**Comment by ahumphreys87** [comment]
Fixed with #2592
